### ISSUE TITLE
Browse form labelling / screen reader announcements

### ISSUE THREAD

**Issue by aviannamiller** [omeka/Omeka#1115]
As originally pointed out in https://github.com/omeka/plugin-Geolocation/issues/78 and moved over here, some browse form elements could use a revision.

1. Issues mainly lie in the "Add a Field" pattern, which adds fields labeled by row numbers. These row numbers can become inaccurate if rows are deleted (so you might get a situation where the remaining rows go from something like row 1 to row 4 to row 6, and so on), and the row numbers themselves are not that helpful as far as accessible names go for these. The term "row" can also imply a table layout, so we should redirect the language here to better indicate both the form and function. We should rethink how we would want to describe each of these fields, maybe think about wrapping them in `fieldset`s, and ensure any numbering or a means of differentiation between similar fields accounts for deleted and added fields.

2. In addition, there's some seemingly overcomplicated `aria-labelledby` values on the "Remove field" button, so it only announces as "Row _x_, button" with no context of the actual button function. The button should still link back to whatever descriptor is decided on for the specific field while also noting the button purpose.

(4.1.3 Status Messages; Level AA; WCAG 2.1)

**Comment by kimisgold** [comment]
Starting work on this in the `a11y/fix-advanced-search-alerts` branch. Would "field" be an acceptable replacement to "row"?

My logic in how I numbered each group of field inputs was to have a persistent number so you would know what would be removed. For example, if you have Fields 1-4 and you deleted the third, you would navigate through Fields 1, 2, and 4, being able to note that 3 is missing as expected. I agree this is tricky. Do we have any good precedents for labeling newly created field groups?
